cobalt/browser/metrics/cobalt_metrics_service_client.cc

```diff
@@ -0,0 +1,196 @@
+}
+
+::metrics::SystemProfileProto::Channel
+CobaltMetricsServiceClient::GetChannel() {
+  // We aren't Chrome and don't follow the same release channel concept.
+  // Return value here is unused in downstream logging.
+  return ::metrics::SystemProfileProto::CHANNEL_UNKNOWN;
+}
+
+std::string CobaltMetricsServiceClient::GetVersionString() {
+  // All version strings will be attached to the GEL event sent by the web
+  // client, so the value doesn't matter much here.
+  return "1.0";
+}
+
+void CobaltMetricsServiceClient::OnEnvironmentUpdate(
+    std::string* serialized_environment) {
+  // Environment updates are serialized SystemProfileProto changes. All
+  // system info should be reported with the web client, so this is a no-op.
+}
+
+void CobaltMetricsServiceClient::CollectFinalMetricsForLog(
+    const base::Closure& done_callback) {
+  // Any hooks that should be called before each new log is uploaded, goes here.
+  // Chrome uses this to update memory histograms. Regardless, you must call
+  // done_callback when done else the uploader will never get invoked.
+  std::move(done_callback).Run();
+}
+
+std::string CobaltMetricsServiceClient::GetMetricsServerUrl() {
+  // Cobalt doesn't upload anything itself, so any URLs are no-ops.
+  return "";
+}
+
+std::string CobaltMetricsServiceClient::GetInsecureMetricsServerUrl() {
+  // Cobalt doesn't upload anything itself, so any URLs are no-ops.
+  return "";
+}
+
+std::unique_ptr<::metrics::MetricsLogUploader>
+CobaltMetricsServiceClient::CreateUploader(
+    base::StringPiece server_url, base::StringPiece insecure_server_url,
+    base::StringPiece mime_type,
+    ::metrics::MetricsLogUploader::MetricServiceType service_type,
+    const ::metrics::MetricsLogUploader::UploadCallback& on_upload_complete) {
+  auto uploader = std::make_unique<CobaltMetricsLogUploader>(
+      service_type, on_upload_complete);
+  log_uploader_ = uploader.get();
+  if (metric_event_handler_ != nullptr) {
+    log_uploader_->SetOnUploadHandler(metric_event_handler_);
+  }
+  return uploader;
+}
+
+base::TimeDelta CobaltMetricsServiceClient::GetStandardUploadInterval() {
+  return base::TimeDelta::FromSeconds(kStandardUploadIntervalSeconds);
+}
+
+bool CobaltMetricsServiceClient::IsReportingPolicyManaged() {
+  // Concept of "managed" reporting policy not applicable to Cobalt.
+  return false;
+}
+
+::metrics::EnableMetricsDefault
+CobaltMetricsServiceClient::GetMetricsReportingDefaultState() {
+  // Metrics always enabled for Cobalt, but this "default state" is unused,
+  // so it's only set to its semantically correct state (checked) out of
+  // principle.
+  return ::metrics::EnableMetricsDefault::OPT_OUT;
+}
+
+bool CobaltMetricsServiceClient::IsUMACellularUploadLogicEnabled() {
+  // Cobalt will never run in a special way for cellular connections.
+  return false;
+}
+
+bool CobaltMetricsServiceClient::SyncStateAllowsUkm() {
+  // UKM currently not used. Value doesn't matter here.
+  return true;
```

Comment:
nit: "false" as the default state would be more expected here. If it's not getting called, better NOTIMPLEMENTED to ensure it does not get called. Here and below

Reply:
Now false